php/class-fm-bylines.php

```diff
@@ -618,6 +678,31 @@ public function get_byline_url( $byline_id, $type = 'author' ) {
 		}
 
 		/**
+		 * Get the byline id associated with a user id
+		 * @param int $user_id
+		 * @return mixed(int|boolean) byline post ID or false.
+		 */
+		public function get_byline_user_mapping( $user_id = null ) {
+			if ( empty( $user_id ) ) {
+				$user_id = get_current_user_id();
+			}
+
+			$bylines = get_posts( array(
+				'post_type' => $this->name,
+				'meta_key' => 'fm_bylines_user_mapping',
+				'meta_value' => absint( $user_id ),
+				'suppress_filters' => false,
+				'ignore_sticky_posts' => true,
+				'no_found_rows' => true,
+			) );
+
+			if ( ! empty( $bylines ) && count( $bylines ) === 1 ) {
+				return $bylines[0]->ID;
+			}
+			return false;
```

Comment:
Do you mean to bail out if more than one post is returned, or do you care about only the first post? If the latter, you could just set `posts_per_page => 1`.